src/main/java/dk/sdu/se_f22/sharedlibrary/db/DBConnection.java

```diff
@@ -38,10 +55,20 @@ private void loadConfig() {
         }
     }
 
-    public static Connection getConnection(){
-        if (dbConnection == null){
-            dbConnection = new DBConnection();
+    public boolean isConnected() {
+        if(connectionPool == null)
+            return false;
+
+        try {
+            var connection = connectionPool.getConnection();
+            return connection.isValid(5);
+        } catch (SQLException e) {
+            return false; // If we get an error, then we can assume that we aren't connected
         }
-        return conn;
+    }
+
+    @Override
+    public Connection getConnection() throws SQLException {
+        return getInstance().connectionPool.getConnection();
```

Comment:
Maybe make the wrapper catch the exception? So users either get a Connection or null, so we don't have to put try-catch everywhere the db is accessed.